examples/stable_diffusion_xl/gm/data/dataset_wds.py

```diff
@@ -248,6 +248,24 @@ def __init__(self, shardlist_desc=None, *args, **kwargs):
         self.dataset = wids.ShardListDataset(shardlist)
         self._datalen = len(self.dataset)
 
+        # preload sample
+        while self.prev_ok_sample is None:
```

Comment:
It's better to check `max_attempts` to avoid dead-loop

Reply:
Fixed